### ISSUE TITLE
CSRF only working on POST methods

### ISSUE THREAD

**Issue by erdlet** [eclipse-ee4j/krazo#89]
With the help of our `HiddenMethodFilter`, we are able to use more HTTP verbs (`PUT`, `PATCH` and `DELETE`) to perfom write actions on resources. Unfortunately, the CSRF protection only works for `POST` requests at the moment.

Currently ,the `CsrfValidateFilter` checks only `@POST` requests for an existing and valid CSRF token. To be able to secure other write actions, we should extend the check to use the other HTTP verbs also. 

This is the part of `CsrfValidateFilter#needsValidation` which is problematic:

```
    private boolean needsValidation(Method controller) {
        if (controller == null || !hasAnnotation(controller, POST.class)) {
            return false;
        }
```

**Comment by chkal** [comment]
Good catch. Yes, we should definitely address this.

**Comment by erdlet** [comment]
I'm working on it at the moment, PR will follow today.

**Comment by erdlet** [comment]
Just saw that this PR has to be based on the open PR #87, because I changed the directory structure of the CSRF tests in this change. So unfortunately we have to merge #87 before the coming fix for this issue.

**Comment by chkal** [comment]
@erdlet I guess this one is fixed by #90, correct?

**Comment by erdlet** [comment]
Thats correct, so we can close this issue.
